conan/tools/microsoft/nmakedeps.py

```diff
@@ -51,8 +51,10 @@ def format_define(define):
                     # https://learn.microsoft.com/en-us/cpp/build/reference/cl-environment-variables
                     macro, value = define.split("=", 1)
                     if value and not value.isnumeric():
-                        value = f'\\"{value}\\"'
+                        value = f'\"{value}\"'
                     define = f"{macro}#{value}"
+                    if " " in define:
+                        define = f"\"{define}\""
```

Comment:
But this will put an extra quotes for strings with spaces, because above has already put quotes for not-numbers?